### PR TITLE
Feature/exam audit it two

### PR DESCRIPTION
**Pull request by chdietz**
Integration tests for audit exam accommodation and item.

### CODE REVIEW

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 1]
```diff
@@ -708,9 +719,6 @@ public void itShouldAuditExamUpdates() {
         jdbcTemplate.update("UPDATE setting s SET s.value = 'TRUE' WHERE s.name = 'AUDIT_TRIGGER_ENABLE';");
         assertThat(countRowsInTableWhere(jdbcTemplate, "setting s", " s.name = 'AUDIT_TRIGGER_ENABLE' and s.value = 'TRUE'")).isEqualTo(1);
 
-        jdbcTemplate.update("INSERT INTO import (id,status,content,contenttype,digest,created) VALUES " +
-                "(-3,0,1, 'application/xml','1D849A91956B74350FF895F067F225E7',CURRENT_DATE);");
-
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: moved to the top

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 2]
```diff
@@ -768,4 +776,215 @@ public void itShouldAuditExamUpdates() {
...
+        accommodations.add(-99);
+        accommodations.add(-96);
+        accommodations.add(-95);
+        repository.upsert(builder.accommodationIds(accommodations).build(), -3);
+
+        // The second update removes 2 accommodations and adds 2 others. (only deletes are audited)
+        assertThat(countRowsInTableWhere(jdbcTemplate, "audit_exam", " exam_id = " + originalExamId)).isEqualTo(2);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "audit_exam_available_accommodation", " exam_id = " + originalExamId)).isEqualTo(2);
+        assertThat(countRowsInTableWhere(jdbcTemplate, "exam_available_accommodation", " exam_id = " + originalExamId)).isEqualTo(3);
+        compareAuditExamAvailableAccommodationToActual(secondExamAccList, originalExamId);
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: helper method to compare saved results before the update with the audit record after the update.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 3]
```diff
@@ -768,4 +776,215 @@ public void itShouldAuditExamUpdates() {
         assertThat(secondExamValues).isEqualTo(secondAuditValues);
     }
 
+    @Test
+    public void itShouldAuditExamAvailableAccommodations() {
+        // todo: Need at one test per audit table where audited record has no null values to ensure no missing columns in trigger sql and audit table.
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: There is one edge case to unsure the IT tests catch a failure to add a column to the trigger sql and/or the audit table.  Will add a test that ensures there are no null values in the record being audited for at least one of each type of audit.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java [hunk 4]
```diff
@@ -768,4 +776,215 @@ public void itShouldAuditExamUpdates() {
...
+                        .responseContentType("type")
+                        .responseDuration(1.1)
+                        .responseTypeId(1)
+                        .subScoreList(subScores)
+                        .administeredAt(Instant.parse("2007-05-02T14:30:00Z"))
+                        .submittedAt(Instant.parse("2007-05-03T14:30:00Z"))
+                        .build())
+        );
+        return items;
+    }
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/repository/impl/ExamRepositoryIT.java]: A method to help build a list of items and specify the 'version' of each item.
